packages/ui/src/fontManager.ts

```diff
@@ -0,0 +1,253 @@
+let currentFontFamily: string = "";
+
+export class FontManager {
+    private static _instance: FontManager;
+
+    private constructor() {}
+
+    public static getInstance(): FontManager {
+        if (!FontManager._instance) {
+            FontManager._instance = new FontManager();
+        }
+        return FontManager._instance;
+    }
+
+    public loadFont(data: ContribFonts): void {
+        // a previous request for loading a font was already made
+        if (this._isFontLoaded(data.fontFamily)) return;
+        try {
+            // override player font style
+            this._overrideCorePlayerFontStyles(data.fontFamily);
+            // making sure no additional calls for loading font will be accepted
+            currentFontFamily = data.fontFamily;
+            // check if download data exists
+            if (!data.downloadData) return;
+            if (data.downloadData.name && data.downloadData.url) {
+                this._handleFontDownloadProcess(data);
+            } else {
+                //download data object is incomplete
+                logger.warn(
+                    "ContribFonts downloadData object should contain both name and url properties.",
+                    {
+                        method: "loadFont"
+                    }
+                );
+            }
+        } catch (err) {
+            logger.warn(
+                `Failed to load font and override core player style with ${data.fontFamily}`,
+                {
+                    method: "loadFont",
+                    data: {
+                        error: err
+                    }
+                }
+            );
+        }
+    }
+
+    private _overrideCorePlayerFontStyles(fontFamily: string): void {
+        const fontCss = `.kaltura-player-container {
+                font-family: inherit;
+            }         
+            .playkit-player {
+                font-family: ${fontFamily};
+            }
+            .playkit-player .playkit-player-gui  {
+                font-family: ${fontFamily};
+            }
+            button, textarea {
+                font-family: inherit;
+            }`;
+        const lastHeadChild = (document.head || document.getElementsByTagName("head")[0])
+            .lastElementChild;
+        const style = document.createElement("style");
+        //adding as last child
+        lastHeadChild.parentNode.insertBefore(style, lastHeadChild.nextSibling);
+        style.appendChild(document.createTextNode(fontCss));
+        logger.info(`Overridden Core player font-family style with: ${fontFamily}`, {
+            method: "_overrideCorePlayerFontStyles"
+        });
+    }
+
+    private _handleFontDownloadProcess(data: ContribFonts): void {
+        const configTestingOptions = data.testingFont ? data.testingFont : {};
+        const fontTestingOptions = ObjectUtils.mergeDeep(
+            DefaultTestingOptions,
+            configTestingOptions
+        ) as TestingFontOptions;
```

Comment:
making the mergeDeep a true typescript method will allow you to remove the as part